### PR TITLE
Implement SavedSignature component

### PR DESCRIPTION
**Pull request by ekoeditaa**
## Context
We need to implement saved signatures feature in PDFTron. 

ClickUp: https://app.clickup.com/t/25523245/UNICORN-75

<a href="https://www.loom.com/share/1f8f8385618a48af8f07272ca9fc5e5d">
    <p>Dedoco Signing App - 21 March 2022 - Watch Video</p>
    
  </a>

## Description
The signing pad we use is from PDFTron. We need to modify the UI components in this repo and then build it to be used in signing-app. This PR adds a new component `SavedSignature` and also implement logic to handle communication between the components using the library.

### CODE REVIEW

File: src/components/SignatureModal/ImageSignature/ImageSignature.js [hunk 1]
```diff
@@ -36,6 +33,11 @@ const ImageSignature = ({
     }
   }, [imageSrc, isTabPanelSelected, isModalOpen]);
 
+  const handleSaveSignatureChange = e => {
+    const newValue = e.target.checked;
+    window.parent.handleSaveSignature?.(newValue);
+  };
+
```
Comment (ekoeditaa) [inline src/components/SignatureModal/ImageSignature/ImageSignature.js]: to set modify state in `signing-app` `PDFWebViewer` element

File: src/components/SignatureModal/ImageSignature/ImageSignature.js [hunk 2]
```diff
@@ -116,66 +122,61 @@ const ImageSignature = ({
...
         )}
       </div>
       <div
         className="footer"
+        style={{
+          justifyContent: 'space-between',
+        }}
       >
-        <button className="signature-create" onClick={createSignature} disabled={!(isModalOpen && isTabPanelSelected) || imageSrc === null || !fileSizeCheck}>
+        <Choice className={`checkbox`} label={'Save Signature'} onChange={handleSaveSignatureChange} />
```
Comment (ekoeditaa) [inline src/components/SignatureModal/ImageSignature/ImageSignature.js]: this is checkbox, it is where the function above is passed to

File: src/components/SignatureModal/InkSignature/InkSignature.js [hunk 3]
```diff
@@ -107,7 +104,12 @@ const InkSignature = ({
     }
     // hack for tool styles for signature not being on state
     forceUpdate();
-  }
+  };
+
+  const handleSaveSignatureChange = e => {
+    const newValue = e.target.checked;
+    window.parent.handleSaveSignature?.(newValue);
+  };
```
Comment (ekoeditaa) [inline src/components/SignatureModal/InkSignature/InkSignature.js:125]: to set modify state in signing-app PDFWebViewer element

File: src/components/SignatureModal/SavedSignature/SavedSignature.js [hunk 4]
```diff
@@ -0,0 +1,156 @@
+import React, { useState, useEffect } from 'react';
```
Comment (ekoeditaa) [inline src/components/SignatureModal/SavedSignature/SavedSignature.js:1]: new component built to contain the saved signatures from server

File: src/components/SignatureModal/SavedSignature/SavedSignature.js [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+  const [t] = useTranslation();
+
+  useEffect(() => {
+    const signatureTool = core.getTool('AnnotationCreateSignature');
+
+    if (isModalOpen && isTabPanelSelected && savedSignatures.length > 0) {
+      var stringLength = savedSignatures[selectedSignatureIndex].length - 'data:image/png;base64,'.length;
+
+      var sizeInBytes = 4 * Math.ceil(stringLength / 3) * 0.5624896334383812;
+      signatureTool.setSignature(savedSignatures[selectedSignatureIndex], sizeInBytes);
```
Comment (ekoeditaa) [inline src/components/SignatureModal/SavedSignature/SavedSignature.js:87]: this is where the signature is created by PDFTron

File: src/components/SignatureModal/SavedSignature/SavedSignature.js [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+    const listener = () => {
+      setSavedSignatures(window.parent?.savedSignatures ?? []);
+    };
+
+    window.parent.addEventListener('savedSignaturesChanged', listener);
+
+    return () => {
+      window.parent.removeEventListener('savedSignaturesChanged', listener);
+    };
+  }, []);
```
Comment (ekoeditaa) [inline src/components/SignatureModal/SavedSignature/SavedSignature.js:101]: this handler is for `signing-app` component to tell this component to re-render this component e.g. when a new signature is saved

**Review by HuyDedoco — APPROVED** [review]

**Review by hieptran51 — APPROVED** [review]

**Pull request merged**